iModelCore/ECDb/ECDb/ECSql/ECSqlPreparer.cpp

```diff
@@ -1511,7 +1534,13 @@ ECSqlStatus ECSqlExpPreparer::PrepareSearchConditionExp(NativeSqlBuilder& native
 ECSqlStatus ECSqlExpPreparer::PrepareSubqueryExp(ECSqlPrepareContext& ctx, SubqueryExp const& exp)
     {
     ctx.GetSqlBuilder().AppendParenLeft();
-    ECSqlStatus stat = ECSqlSelectPreparer::Prepare(ctx, *exp.GetQuery());
+    SelectStatementExp const* selectSubquery =  exp.GetQuery<SelectStatementExp>();
+    CommonTableExp const* cteSubquery =  exp.GetQuery<CommonTableExp>();

```

Comment:
same here, it's better to get the cte right before you check for null

Reply:
fixed